### PR TITLE
Transition to manifold - Part 1 - Examples 1 to 16

### PR DESCRIPTION
**Pull request by luca-heltai**
Before I move forward to next bunch of examples, I wanted to make sure you all agreed on this, since I'm changing pretty ancient history here...

;)

Addresses in part #339

### CODE REVIEW

File: examples/step-1/step-1.cc [hunk 1]
```diff
@@ -99,31 +100,40 @@ void second_grid ()
...
-  // and given by the cells of the coarse grid (which we just created). It
-  // uses this information when cells at the boundary are refined and new
-  // points need to be introduced on the boundary; if the boundary is assumed
-  // to be straight, then new points will simply be in the middle of the
+  // By default, the triangulation assumes that all boundaries are
+  // straigth lines, and all cells are bi-linear quads, and that they
+  // are defined by the cells of the coarse grid (which we just
+  // created). It uses this information when cells are refined and new
+  // points need to be introduced; if the domain is assumed to be
+  // flat, then new points will simply be in the middle of the
```
Comment (bangerth) [inline examples/step-1/step-1.cc]: I think the word "flat" is maybe a bit out of place here because the reader is not thinking of this here in the context of manifolds etc. How about rewording this as follows:
```
// By default, the triangulation assumes that all boundaries are
+ // straight lines, and all cells are bi-linear quads, and that they
+ // are defined by the cells of the coarse grid (which we just
+ // created). Unless we do something special, when new
+ // points need to be introduced; the domain is assumed to be
+ // delineated by the straight lines of the coarse mesh, and new points will simply be in the middle of the
```

File: examples/step-1/step-1.cc [hunk 2]
```diff
@@ -99,31 +100,40 @@ void second_grid ()
...
+  // Here, however, we know that the domain is curved, and we would
+  // like to have the Triangulation place new points according to the
+  // underlying geometry. Fortunately, some good soul implemented an
+  // object which describes a spherical domain, of which the ring is a
+  // section; it only needs the center of the ring and automatically
+  // figures out how to instruct the Triangulation where to place the
+  // new points. We first set the "manifold indicator" of all cells
+  // and faces of the Triangulation to the value zero, and then
+  // associate the curved Manifold object with those parts of the
+  // Triangulation that have the "manifold indicator" zero. By
```
Comment (bangerth) [inline examples/step-1/step-1.cc]: Instead of this sentence, maybe do something like this: "[...to place the new points.] The way this works in deal.II is that you tag parts of the triangulation you want to be curved with a number that is usually referred to as "boundary indicator" and then tell the triangulation to use a particular "manifold object" for all places with this manifold indicator. How exactly this works is not important at this point (you can read up on it in step-53 and @ref manifold). Here, for simplicity, we will choose the manifold id to be zero. [By default...]"

File: examples/step-1/step-1.cc [hunk 3]
```diff
@@ -99,31 +100,40 @@ void second_grid ()
...
+  // default, all cells and faces of the Triangulation have their
+  // manifold_id set to numbers::invalid_manifold_id, which is the
+  // default if you want a flat manifold, but you can change this
+  // number for individual cells and faces. In that case, the curved
+  // manifold thus associated with number zero will not apply on those
```
Comment (bangerth) [inline examples/step-1/step-1.cc]: apply TO those

File: examples/step-1/step-1.cc [hunk 4]
```diff
@@ -99,31 +100,40 @@ void second_grid ()
...
+  // parts with a non-zero manifold indicator, but other manifold
+  // description objects can be associated with those non-zero
+  // indicators. If no manifold description is associated with a
+  // particular manifold indicator, a flat manifold is
```
Comment (bangerth) [inline examples/step-1/step-1.cc]: again, maybe replace "flat manifold" by "a manifold that produces straight edges"?

File: examples/step-10/step-10.cc [hunk 5]
```diff
@@ -77,12 +77,12 @@ namespace Step10
 
     // So first generate a coarse triangulation of the circle and associate a
     // suitable boundary description to it. Note that the default values of
-    // the HyperBallBoundary constructor are a center at the origin and a
-    // radius equals one.
+    // the SphericalManifold constructor is a center at the origin.
```
Comment (bangerth) [inline examples/step-10/step-10.cc]: in the line before the one you changes, values->value

Comment (bangerth) [inline examples/step-10/step-10.cc]: maybe say "the default value of the argument to"

**Comment by bangerth** [comment]
With minor changes, this is ok, but I have a question: when you run all of these changed tutorial programs, you should get different results because previously the spherical boundary was used only on the boundary but now you do it for all cells. That's ok, I suppose, but the pictures that come out and the numerical results on the screen should change as well, don't they?

The direction is definitely right. I hadn't realized how many of our programs run on the circle -- from a didactic perspective, this may not be the best idea, but I suppose it is what it is.

**Comment by luca-heltai** [comment]
I have only changed the *interior* of the domain for step-1, not for all the examples, since all the other ones I corrected are *circles*, and not hypershells. I didn't get yet to those examples that run on hyper-shells, but there I'll make sure to also change the pictures.

Here is the image for grid-2 now:

grid-2

**Comment by bangerth** [comment]
Ah, I had missed that -- thanks for pointing it out. Please fix the one issue (plus another occurrence of the word "flat" on line 132 of step-1) and merge. 

What should we do with the new picture? Just overwrite the old one and live with the fact that the 8.2 docs (and from previous releases) then point to the wrong figure? Other ideas?

**Comment by tamiko** [comment]
> What should we do with the new picture?

What about giving it a revision number, e.g. name-r2.png, and just point
to that?

**Comment by luca-heltai** [comment]
I've modified the name of the image to step-1.grid-2r2.png, and searched through the entire step-1.cc replacing some more occurences of "boundary indicator" and "boundary object" with "manifold indicator" and "manifold object". 

The other steps should be fine, since in those steps we are only modifying the boundary, and it still makes sense to talk about "boundary object".

**Comment by bangerth** [comment]
Nice work, Luca -- many thanks!

I've also uploaded the picture to the website, so it should link to alright.
